xbmc/interfaces/json-rpc/PVROperations.cpp

```diff
@@ -282,8 +283,12 @@ JSONRPC_STATUS CPVROperations::Scan(const std::string &method, ITransportLayer *
   if (!CServiceBroker::GetPVRManager().IsStarted())
     return FailedToExecute;
 
-  if (CServiceBroker::GetPVRManager().GUIActions()->StartChannelScan(
-      parameterObject["clientid"].asInteger()))
+  int clientId = PVR_INVALID_CLIENT_ID;
+
+  if (parameterObject.isMember("clientid"))
+    clientId = parameterObject["clientid"].asInteger();
+
+  if (CServiceBroker::GetPVRManager().GUIActions()->StartChannelScan(clientId))
```

Comment:
if you would introduce GUIActions::StartChannelScan() without params, you would eliminate the dependency to PVRClient.h

Reply:
Makes perfect sense.